Add migration guide from v1.x to v2.x (#4850)
FerretDB/FerretDB: website/docs/migration/migrate-from-v1.md

Reviewer: CI is can't build this page https://github.com/FerretDB/FerretDB/actions/runs/13645489615/job/38143550418?pr=4850 due to broken links

```
Error:  Error: Unable to build website for locale en.
[INFO] Docusaurus version: 3.7.0
Node version: v22.13.0
    at tryToBuildLocale (/workdir/docusaurus-docs/node_modules/@docusaurus/core/lib/commands/build/build.js:78:15)
    at async /workdir/docusaurus-docs/node_modules/@docusaurus/core/lib/commands/build/build.js:34:9
    ... 4 lines matching cause stack trace ...
    at async file:///workdir/docusaurus-docs/node_modules/@docusaurus/core/bin/docusaurus.mjs:44:3 {
  [cause]: Error: Docusaurus found broken links!
  
  Please check the pages of your site in the list below, and make sure you don't reference any path that does not exist.
  Note: it's possible to ignore broken links with the 'onBrokenLinks' Docusaurus configuration, and let the build pass.
  
  Exhaustive list of all broken links found:
  - Broken link on source page path = /migration/migrate-from-v1/:
     -> linking to ../installation/documentdb/ (resolved as: /migration/installation/documentdb/)
  
      at throwError (/workdir/docusaurus-docs/node_modules/@docusaurus/logger/lib/logger.js:80:11)
      at reportBrokenLinks (/workdir/docusaurus-docs/node_modules/@docusaurus/core/lib/server/brokenLinks.js:250:47)
      at handleBrokenLinks (/workdir/docusaurus-docs/node_modules/@docusaurus/core/lib/server/brokenLinks.js:282:5)
      at executeBrokenLinksCheck (/workdir/docusaurus-docs/node_modules/@docusaurus/core/lib/commands/build/buildLocale.js:91:47)
      at /workdir/docusaurus-docs/node_modules/@docusaurus/core/lib/commands/build/buildLocale.js:70:67
      at Object.async (/workdir/docusaurus-docs/node_modules/@docusaurus/logger/lib/perfLogger.js:42:47)
      at buildLocale (/workdir/docusaurus-docs/node_modules/@docusaurus/core/lib/commands/build/buildLocale.js:70:31)
      at async runBuildLocaleTask (/workdir/docusaurus-docs/node_modules/@docusaurus/core/lib/commands/build/build.js:93:5)
      at async /workdir/docusaurus-docs/node_modules/@docusaurus/core/lib/commands/build/build.js:74:13
      at async tryToBuildLocale (/workdir/docusaurus-docs/node_modules/@docusaurus/core/lib/commands/build/build.js:70:9)
      at async /workdir/docusaurus-docs/node_modules/@docusaurus/core/lib/commands/build/build.js:34:9
      at async mapAsyncSequential (/workdir/docusaurus-docs/node_modules/@docusaurus/utils/lib/jsUtils.js:21:24)
      at async Command.build (/workdir/docusaurus-docs/node_modules/@docusaurus/core/lib/commands/build/build.js:33:5)
      at async Promise.all (index 0)
      at async runCLI (/workdir/docusaurus-docs/node_modules/@docusaurus/core/lib/commands/cli.js:56:5)
      at async file:///workdir/docusaurus-docs/node_modules/@docusaurus/core/bin/docusaurus.mjs:44:3
}
task: Failed to run task "docs": exit status 1
Error: Process completed with exit code 201.
```

---
sidebar_position: 3
---

> # Migrating from FerretDB v1.x

Reviewer: Do we mention anywhere that for documentdb extension to initialize in docker, the database has to be empty? I think we had a few query about documentdb extension not being installed.

Author: Thanks for mentioning this. It should probably go into installation guide since  they both will cover the actual initialization in docker. 

It could also be an entry into the troubleshooting guide.

> # Migrating from FerretDB v1.x

Reviewer: Could you keep the namings of header consistent with other heading? Also the file name too.

See "Migrat**ion** from MongoDB" and "Migrat**ing** from FerretDB v1.x"
```suggestion
# Migrating from FerretDB v1.x
```

![image](https://github.com/user-attachments/assets/9bd2d985-a82c-440f-aefa-f65c72772862)

Reviewer: Do we want file name to match the heading?

Reviewer 2: File name is being used as a path in the URL, and we want the URL to match the page name

If you are running an older version of FerretDB (v1.24 or earlier versions), we recommend upgrading to the latest version.

These are the major differences between FerretDB v1.x and v2.x:

1. Unlike v1.x that provides options for PostgreSQL and SQLite as backend, FerretDB v2.x **requires a PostgreSQL with DocumentDB extension** as the backend.
   Find the [installation guide for PostgreSQL with DocumentDB extension here](../installation/documentdb/docker.md).

2. **`PLAIN` authentication is no longer supported** in FerretDB v2.x.
   FerrretDB v2.x uses `SCRAM-SHA-256` for authentication.

>    So if your connection string in v1.x looks like `mongodb://username:password@localhost:27017/ferretdb?authMechanism=PLAIN`, use`mongodb://username:password@localhost:27017/` with 2.x.

Reviewer: Need space

```suggestion
   So if your connection string in v1.x looks like `mongodb://username:password@localhost:27017/ferretdb?authMechanism=PLAIN`, use `mongodb://username:password@localhost:27017/` with 2.x.
```

>    So if your connection string in v1.x looks like this (`mongodb://username:password@localhost:27017/ferretdb?authMechanism=PLAIN`), in v2.x, (`mongodb://username:password@localhost:27017/`).

Reviewer 2: ```suggestion
   So if your connection string in v1.x looks like `mongodb://username:password@localhost:27017/ferretdb?authMechanism=PLAIN`, use`mongodb://username:password@localhost:27017/` with 2.x.
```

Brackets look odd, and a word seem to be missing

   [Find out more on FerretDB authentication here](../security/authentication.md).

## Backup and restore data from FerretDB v1.x to v2.x

The migration process follows the same steps as MongoDB to FerretDB.

> Follow [this guide to dump and restore your data](migrating-from-mongodb.md).

Reviewer 2: So is it "dump" or "backup"?
Also, the title is odd: it looks like we are backing up data from 1.x to 2.x, using the latter as a backup.
